compiler/shared/main/scala/mlscript/compiler/mono/Monomorph.scala

```diff
@@ -249,7 +250,7 @@ class Monomorph(debug: Debug = DummyDebug) extends DataTypeInferer:
 
   private def nuUpdateFunction(funcName: String): Unit = {
     val updateCount = nuEvalCnt.get(funcName).getOrElse(0)
-    if(updateCount <= 10){
+    if(updateCount <= 3){
```

Comment:
So the stack depth limit is... 3?

Reply:
Was a temporary change while debugging a recursion bug.